keystone_plugin/project.py

```diff
@@ -119,21 +113,32 @@ def validate_users(users, keystone_client):
         keystone_client.users.find(name=user_name)
 
     for user in users:
-        if len(user['roles']) > len(set(user['roles'])):
+        if len(user[ROLES]) > len(set(user[ROLES])):
             msg = 'Roles for user {} are not unique'
             raise NonRecoverableError(msg.format(user['name']))
 
-    role_names = {role for user in users for role in user['roles']}
+    role_names = {role for user in users for role in user[ROLES]}
     for role_name in role_names:
         keystone_client.roles.find(name=role_name)
 
 
+def get_quota(tenant_id, client, what_quota):
+    if what_quota == NEUTRON:
+        quota = client.get_quotas_tenant(tenant=tenant_id)
+        return quota
+    else:
+        quota = client.quotas.get(tenant_id=tenant_id)
```

Comment:
I would like to see some exception handling, either logging or raising what happened here. For this one, we can probably get away with ignoring a not found error and just log it as an error.